### PR TITLE
[PoliciesConfig] validate json

### PR DESCRIPTION
**Pull request by macejmic**
**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-1059

### CODE REVIEW

File: test/integration/user-management-api/services/proxy/policies_test.rb [hunk 1]
```diff
@@ -58,6 +58,12 @@ def test_invalid_policies_config
     assert_response :unprocessable_entity
   end
 
+  def test_invalid_json_policies_config
+    put admin_api_service_proxy_policies_path(valid_params.merge({ proxy: { policies_config: { name: 'echo '}.to_json }}))
+    binding.pry
```
Comment (Martouta) [inline test/integration/user-management-api/services/proxy/policies_test.rb:63]: 👀

File: test/integration/user-management-api/services/proxy/policies_test.rb [hunk 2]
```diff
@@ -58,6 +58,11 @@ def test_invalid_policies_config
     assert_response :unprocessable_entity
   end
 
+  def test_invalid_json_policies_config
+    put admin_api_service_proxy_policies_path(valid_params.merge({ proxy: { policies_config: { name: 'echo '}.to_json }}))
+    assert_response :unprocessable_entity
+  end
```
Comment (Martouta) [inline test/integration/user-management-api/services/proxy/policies_test.rb:64]: This should also test that the response.body contains the proper error message in policies_config

Comment (macejmic) [reply test/integration/user-management-api/services/proxy/policies_test.rb]: done.

File: app/models/proxy.rb [hunk 3]
```diff
@@ -559,12 +560,16 @@ def proxy
   class PolicyConfig
     include ActiveModel::Validations
 
+    attr_reader :init_attributes
     attr_accessor :name, :version, :configuration, :enabled
-    validates :name, :version, presence: true
-    validate :configuration_is_object
+
+    validates :name, :version, presence: true, if: :attributes_are_hash?
```
Comment (hallelujah) [inline app/models/proxy.rb:566]: Why not raise an error if passed a wrong argument?

File: app/models/proxy.rb [hunk 4]
```diff
@@ -559,12 +560,16 @@ def proxy
...
+    validate :configuration_is_object, if: :attributes_are_hash?
+    validate :attributes_are_hash
 
     def initialize(attributes = {})
-      self.attributes = attributes
+      @init_attributes = attributes
```
Comment (hallelujah) [inline app/models/proxy.rb:571]: `#initialize` should check the argument passed instead of validating them, wdyt?

File: app/models/proxy.rb (hunk 3 above)
Comment (macejmic) [reply app/models/proxy.rb:566]: QE is complaining about 500.. so that wouldn't solve the problem.

File: app/models/proxy.rb (hunk 4 above)
Comment (macejmic) [reply app/models/proxy.rb:571]: okok, that could be also the way.

**Pull request merged**